common/maplib/move-rule.mjs

```diff
@@ -0,0 +1,51 @@
+import {createRequire} from 'module';
+const require = createRequire(import.meta.url);
+
+const config = require('config');
+const movingRequestThreshold = config.get('movingRequestThreshold');
+
+class MoveRule{
+
+  _getManhattanDistance(a,b){
+    return Math.abs(a.x - b.x) + Math.abs(a.y - b.y);
+  }
+
+  sameMapCheck(mapCoord1,mapCoord2){
```

Comment:
Please add brief method comment. Same below.

Reply:
If you implement `MapCoord.distanceTo()` as I stated above, this function is not needed.